nico/utils.nim

```diff
@@ -162,11 +162,17 @@ proc richPrint*(text: string, x,y: int, align: TextAlign = taLeft, shadow: bool
         elif code == "s":
           wiggle = true
         elif code.startsWith("spr"):
-          var sprId: int
+          var sprId, palA, palB: int
           var (sw,sh) = spriteSize()
+
           if scanf(code, "spr($i,$i,$i)", sprId, sw, sh):
             spr(sprId, x - (if align == taCenter: tlen div 2 elif align == taRight: tlen else: 0), y)
             x += sw
+          elif scanf(code, "spr($i)pal($i,$i)", sprId, palA, palB):
+            pal(palA, palB)
+            spr(sprId, x - (if align == taCenter: tlen div 2 elif align == taRight: tlen else: 0), y)
+            pal()
```

Comment:
this will reset all palette mappings, would be best to only reverse the change just made